### ISSUE TITLE
Unrecognized seccomp action "notify"

### ISSUE THREAD

**Issue by GinesGarcia** [lxc/lxc#3095]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 18.04
 * The output of
   * `lxc-start --version`  - 3.1.0-devel
   * `lxc-checkconfig`

```
--- Namespaces ---
Namespaces: enabled
Utsname namespace: enabled
Ipc namespace: enabled
Pid namespace: enabled
User namespace: enabled
Network namespace: enabled

--- Control groups ---
Cgroups: enabled

Cgroup v1 mount points:
/sys/fs/cgroup/systemd 
/sys/fs/cgroup/memory  
/sys/fs/cgroup/devices 
/sys/fs/cgroup/rdma    
/sys/fs/cgroup/pids    
/sys/fs/cgroup/hugetlb 
/sys/fs/cgroup/cpu,cpuacct
/sys/fs/cgroup/perf_event
/sys/fs/cgroup/net_cls,net_prio
/sys/fs/cgroup/freezer
/sys/fs/cgroup/cpuset
/sys/fs/cgroup/blkio

Cgroup v2 mount points:
/sys/fs/cgroup/unified
Cgroup v2 mount points:
/sys/fs/cgroup/unified

Cgroup v1 clone_children flag: enabled
Cgroup device: enabled                
Cgroup sched: enabled                 
Cgroup cpu account: enabled
Cgroup memory controller: enabled
Cgroup cpuset: enabled

--- Misc ---
Veth pair device: enabled, loaded
Macvlan: enabled, not loaded
Vlan: enabled, not loaded
Bridges: enabled, not loaded
Advanced netfilter: enabled, not loaded
CONFIG_NF_NAT_IPV4: enabled, loaded
CONFIG_NF_NAT_IPV6: enabled, loaded
CONFIG_IP_NF_TARGET_MASQUERADE: enabled, not loaded
CONFIG_IP6_NF_TARGET_MASQUERADE: enabled, not loaded
CONFIG_NETFILTER_XT_TARGET_CHECKSUM: enabled, not loaded
CONFIG_NETFILTER_XT_MATCH_COMMENT: enabled, not loaded
FUSE (for use with lxcfs): enabled, not loaded

--- Checkpoint/Restore ---
checkpoint restore: enabled
CONFIG_FHANDLE: enabled
CONFIG_EVENTFD: enabled
CONFIG_EPOLL: enabled
CONFIG_UNIX_DIAG: enabled
CONFIG_INET_DIAG: enabled
CONFIG_PACKET_DIAG: enabled
CONFIG_NETLINK_DIAG: enabled
File capabilities:

Note : Before booting a new kernel, you can check its configuration
usage : CONFIG=/path/to/config /usr/local/bin/lxc-checkconfig
```
   * `uname -a`
Linux lxd-testing 4.15.0-54-generic
   * `cat /proc/self/cgroup`
```
12:blkio:/user.slice
11:cpuset:/
10freezer:/
9:net_cls,net_prio:/
8:perf_event:/
7:cpu,cpuacct:/user.slice
6:hugetlb:/
5:pids:/user.slice/user-1000.slice/session-3.scope
4:rdma:/
3:devices:/user.slice
2:memory:/user.slice
1:name=systemd:/user.slice/user-1000.slice/session-3.scope
0::/user.slice/user-1000.slice/session-3.scope
```
   * `cat /proc/1/mounts`

# Issue description
When using seccomp config I can use errno directives over syscalls but when activating "notify" directive, I get "Unrecognized seccomp action "notify" error.

A brief description of what failed or what could be improved.

# Steps to reproduce
 1. Step one
git clone https://github.com/lxc/lxc.git
 2. Step two
cd lxc; ./autogen.sh && ./configure --enable-seccomp && make && make install
 3. Step three
add "fork notify" directive to  /usr/local/share/lxc/config/common.seccomp
 4. Step four
sudo lxc-start container_name

**Comment by brauner** [comment]
libseccomp on your system does not support notify. It hasn't been release yet I fear.

**Comment by GinesGarcia** [comment]
Hi,
thanks a lot for the reply.

What do you mean with "system"? is it the operating system?

thanks in advance,
Ginés.

**Comment by brauner** [comment]
Yes, the libseccom package in 18.04 on Ubuntu does not ship with notify support. The upstream project for libseccomp has also not yet made a release that supports the notify patches that lxc carries.

**Comment by GinesGarcia** [comment]
oh!
well, thanks a lot for the reply!

**Comment by GinesGarcia** [comment]
One quick question,
do you know which OS should I use in order to have notify features available?

Regards,

**Comment by brauner** [comment]
On Thu, Jul 18, 2019 at 1:46 PM Ginés García Avilés <
notifications@github.com> wrote:

> One quick question,
> do you know which OS should I use in order to have notify features
> available?
>
> Regards,
>

There is none where this feature is available since the libseccomp project
has not made release
with that feature which distributions could've picked up.

**Comment by GinesGarcia** [comment]
ok,
I've been checking a little bit libseccomp and those features are already available (under kernel 5.0 on Ubuntu). The question is, what can I do at LXC in order to be able to use that functionality? is it feasible? or on the other hand, is it independent of LXC implementation?
